### PR TITLE
Add REMOTE_AUTH_AUTOCREATE_GROUPS

### LINKED ISSUE #7671: Auto-create remote groups
### NetBox version

v3.0.8

### Feature type

Change to existing functionality

### Proposed functionality

Allow NetBox to auto-create groups referenced in the `REMOTE_AUTH_GROUP_HEADER`.

### Use case

As a data center administrator using Keycloak to manage auth/authz, I would like NetBox to automatically instantiate groups provided by Keycloak when using `REMOTE_AUTH_ENABLED`.

With `REMOTE_AUTH_ENABLED` and `REMOTE_AUTH_GROUP_SYNC_ENABLED`, it's still necessary to manually create groups before the group sync is effective.  I would like to see a `REMOTE_AUTH_AUTO_CREATE_GROUPS` setting, analagous to `REMOTE_AUTH_AUTO_CREATE_USER` but for groups, that would cause NetBox to automatically create any groups in the `REMOTE_AUTH_GROUP_HEADER`. This would allow authorization to be completely delegated to the identity management platform.

This is the same feature requested by [this comment](https://github.com/netbox-community/netbox/issues/5775#issuecomment-883651758) in #5775.

### Database changes

_No response_

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by larsks**
### Closes: #7671

When REMOTE_AUTH_AUTOCREATE_GROUPS is True, Netbox will create groups
referenced in the REMOTE_AUTH_GROUP_HEADER that don't exist in the
database.

### CODE REVIEW

**Comment by candlerb** [comment]
The linked issue is #7671 (although not yet accepted)

**Comment by larsks** [comment]
Sorry, it was linked in the commit message but not the PR description.

**Comment by larsks** [comment]
I've rebased this on the current `develop` branch.

File: netbox/netbox/settings.py [hunk 1]
```diff
@@ -102,6 +102,7 @@
 PLUGINS_CONFIG = getattr(configuration, 'PLUGINS_CONFIG', {})
 RELEASE_CHECK_URL = getattr(configuration, 'RELEASE_CHECK_URL', None)
 REMOTE_AUTH_AUTO_CREATE_USER = getattr(configuration, 'REMOTE_AUTH_AUTO_CREATE_USER', False)
+REMOTE_AUTH_AUTO_CREATE_GROUP = getattr(configuration, 'REMOTE_AUTH_AUTO_CREATE_GROUP', False)
```
Comment (kkthxbye-code) [inline netbox/netbox/settings.py:105]:
```suggestion
REMOTE_AUTH_AUTO_CREATE_GROUPS = getattr(configuration, 'REMOTE_AUTH_AUTO_CREATE_GROUPS', False)
```

I think you made a copy paste error. When changing it to the above suggestion the PR works as intended.

Comment (kkthxbye-code) [reply netbox/netbox/settings.py:105]: You should also add the new setting to the docs.

**Review by kkthxbye-code — CHANGES_REQUESTED** [review]

**Comment by kkthxbye-code** [comment]
@larsks - are you still interested in finishing this? When my review has been addressed, it should be good to go.

**Comment by larsks** [comment]
@kkthxbye-code yes, still interested; just ended up with my head in other things for a bit. I'll try submitting an update over the next couple of days.

**Comment by davama** [comment]
@larsks 
Hope you are well. Would be great to have this feature!
I can help in testing if need be
Thank you for wiping this up

Regards

**Pull request closed without merging**